### DISCUSSION TITLE
Using the native-micro Dockerfile with automatic building

### DISCUSSION THREAD

**Discussion by bulldog98** [quarkusio/quarkus#33406] [Q&A] [answered]
I'm using the following native compile settings
```
quarkus.container-image.build=true
quarkus.package.type=native
quarkus.native.container-build=true
quarkus.docker.dockerfile-native-path=${project.root}/src/main/docker/Dockerfile.native-micro
```

Problem is that
a) the [documenation](https://quarkus.io/guides/container-image#DockerOptions) says that the dockerfile-native-path is for jvm
b) the build still tries to use the `${project.root}/src/main/docker/Dockerfile.native`  to build the container

Is that intended and how to use the native-micro Dockerfile on container-image build?

**Comment by machi1990** [comment]
> a) the [documenation](https://quarkus.io/guides/container-image#DockerOptions) says that the dockerfile-native-path is for jvm

The documentation is wrong and it should be fixed here https://github.com/quarkusio/quarkus/blob/67644b6b5f33ec4ddd6e2c801fc5e977e2b8fab4/extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerConfig.java#L26
It should be `native` and not `JVM`.  A PR is welcome. 

> Is that intended and how to use the native-micro Dockerfile on container-image build?

If the value is not set, the default one which is `${project.root}/src/main/docker/Dockerfile.native` will be used.

**Reply by bulldog98** [reply]
Will do a PR this evening. Problem for me is mainly that the settings I listed in the question are in my application.properties and quarkus still tries to build with the default Dockerfile.

**Reply by machi1990** [reply]
@bulldog98 thanks. The configuration knobs shared looks correct to me. And what steps are you using to build the native image? @geoand would you be able to tell what could be wrong here?

**Reply by geoand** [reply]
It seems correct at first glance - I would need to check

**Reply by bulldog98** [reply]
I was able to reproduce that with [code-with-quarkus.zip](https://github.com/quarkusio/quarkus/files/11487435/code-with-quarkus.zip)

**Reply by geoand** [reply]
I think I know what's going on, but I'll have to check

**Reply by geoand** [reply, marked as answer]
As I expected, you need to use 

```xml
    <resources>
      <resource>
        <directory>src/main/resources</directory>
        <filtering>true</filtering>
      </resource>
    </resources>
```

in the `<build>` section of the `pom.xml` file. This allows Maven to replace the placeholders. 

However, you are also using the wrong Maven variable, you need to use `${project.basedir}` instead of `${project.root}`

**Reply by bulldog98** [reply]
Is there a analog requirement for gradle?

**Reply by geoand** [reply]
There are ways to do the same thing in Gradle but I don't know them off hand

**Reply by bulldog98** [reply]
Hm I got the syntax also from the javadoc, now I'm not sure what should really be in that.

**Reply by geoand** [reply]
Good point. It's meant to be used as a placeholder, but let me see if I can rephrase it somehow.
